plugins/repository-s3/src/main/java/org/opensearch/repositories/s3/async/AsyncTransferManager.java

```diff
@@ -129,6 +129,7 @@ private void uploadInParts(
 
         CreateMultipartUploadRequest.Builder createMultipartUploadRequestBuilder = CreateMultipartUploadRequest.builder()
             .bucket(uploadRequest.getBucket())
+            .metadata(uploadRequest.getMetadata())
```

Comment:
what if the upload request has the metadata as null or empty?

Reply:
We don't provide metadata in Request Builder if metadata is empty. Updated.